### ISSUE TITLE
Over Reads on a Topic

### ISSUE THREAD

**Issue by GeminPatel** [zendesk/racecar#57]
Hi,

I started a consumer subscribing to an empty topic. Then with extreme speed dumped 37869 messages in a that clean topic. My consumer side by side started fetching. Looking at the log shows that it over processed. It processed till 37904 and then reset back to the correct one. Why did this happen? And this time my count of processes also reduced to 37756 instead of 37869.

Log:
```

D, [2018-02-01T19:20:55.952360 #13490] DEBUG -- : Marking users/0:37990 as processed
D, [2018-02-01T19:20:55.953482 #13490] DEBUG -- : Marking users/0:37868 as processed
D, [2018-02-01T19:20:55.956459 #13490] DEBUG -- : Marking users/0:37869 as processed
D, [2018-02-01T19:20:55.957973 #13490] DEBUG -- : Marking users/0:37870 as processed
D, [2018-02-01T19:20:55.959402 #13490] DEBUG -- : Marking users/0:37871 as processed
D, [2018-02-01T19:20:55.960880 #13490] DEBUG -- : Marking users/0:37872 as processed
D, [2018-02-01T19:20:55.962321 #13490] DEBUG -- : Marking users/0:37873 as processed
D, [2018-02-01T19:20:55.963735 #13490] DEBUG -- : Marking users/0:37874 as processed
D, [2018-02-01T19:20:55.965176 #13490] DEBUG -- : Marking users/0:37875 as processed
D, [2018-02-01T19:20:55.966594 #13490] DEBUG -- : Marking users/0:37876 as processed
D, [2018-02-01T19:20:55.968016 #13490] DEBUG -- : Marking users/0:37877 as processed
D, [2018-02-01T19:20:55.969427 #13490] DEBUG -- : Marking users/0:37878 as processed
D, [2018-02-01T19:20:55.970775 #13490] DEBUG -- : Marking users/0:37879 as processed
D, [2018-02-01T19:20:55.972231 #13490] DEBUG -- : Marking users/0:37880 as processed
D, [2018-02-01T19:20:55.975846 #13490] DEBUG -- : Marking users/0:37881 as processed
D, [2018-02-01T19:20:55.978254 #13490] DEBUG -- : Marking users/0:37882 as processed
D, [2018-02-01T19:20:55.980070 #13490] DEBUG -- : Marking users/0:37883 as processed
D, [2018-02-01T19:20:55.981492 #13490] DEBUG -- : Marking users/0:37884 as processed
D, [2018-02-01T19:20:55.982938 #13490] DEBUG -- : Marking users/0:37885 as processed
D, [2018-02-01T19:20:55.984352 #13490] DEBUG -- : Marking users/0:37886 as processed
D, [2018-02-01T19:20:55.985751 #13490] DEBUG -- : Marking users/0:37887 as processed
D, [2018-02-01T19:20:55.987158 #13490] DEBUG -- : Marking users/0:37888 as processed
D, [2018-02-01T19:20:55.988612 #13490] DEBUG -- : Marking users/0:37889 as processed
D, [2018-02-01T19:20:55.990045 #13490] DEBUG -- : Marking users/0:37890 as processed
D, [2018-02-01T19:20:55.991432 #13490] DEBUG -- : Marking users/0:37891 as processed
D, [2018-02-01T19:20:55.992823 #13490] DEBUG -- : Marking users/0:37892 as processed
D, [2018-02-01T19:20:55.994276 #13490] DEBUG -- : Marking users/0:37893 as processed
D, [2018-02-01T19:20:55.995800 #13490] DEBUG -- : Marking users/0:37894 as processed
D, [2018-02-01T19:20:56.004625 #13490] DEBUG -- : Marking users/0:37895 as processed
D, [2018-02-01T19:20:56.007385 #13490] DEBUG -- : Marking users/0:37896 as processed
D, [2018-02-01T19:20:56.008876 #13490] DEBUG -- : Marking users/0:37897 as processed
D, [2018-02-01T19:20:56.010345 #13490] DEBUG -- : Marking users/0:37898 as processed
D, [2018-02-01T19:20:56.012165 #13490] DEBUG -- : Marking users/0:37899 as processed
D, [2018-02-01T19:20:56.014249 #13490] DEBUG -- : Marking users/0:37900 as processed
D, [2018-02-01T19:20:56.016525 #13490] DEBUG -- : Marking users/0:37901 as processed
D, [2018-02-01T19:20:56.017950 #13490] DEBUG -- : Marking users/0:37902 as processed
D, [2018-02-01T19:20:56.019486 #13490] DEBUG -- : Marking users/0:37903 as processed
D, [2018-02-01T19:20:56.021068 #13490] DEBUG -- : Marking users/0:37904 as processed
D, [2018-02-01T19:20:56.021225 #13490] DEBUG -- : Fetching batch from users/0 starting at offset 37905
D, [2018-02-01T19:20:56.021392 #13490] DEBUG -- : Sending fetch API request 35 to 10.10.10.10:9092
D, [2018-02-01T19:20:56.021595 #13490] DEBUG -- : Waiting for response 35 from 10.10.10.10:9092
D, [2018-02-01T19:20:56.022878 #13490] DEBUG -- : Received response 35 from 10.10.10.10:9092
E, [2018-02-01T19:20:56.024361 #13490] ERROR -- : Invalid offset for users/0, resetting to default offset
D, [2018-02-01T19:20:56.024567 #13490] DEBUG -- : Sending list_offset API request 36 to 10.10.10.10:9092
```

Consumer:
```

class TestConsumer < Racecar::Consumer
  subscribes_to 'users', start_from_beginning: false

  def process(message)
    count = Rails.cache.read("count")

    unless count
      count = 0
    end

    Rails.cache.write("count", count + 1)
  end
end
```

Versions:

- Racecar: 0.3.5, 0.3.7 [tried on both]
- Ruby Kafka: 0.5.1
- Ruby: 2.3.1

**Comment by GeminPatel** [comment]
PS: This is fixed in Ruby Kafka: 0.5.2
The overread aswell as count

**Comment by dasch** [comment]
Yup.
